clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1913,6 +1923,7 @@ private boolean processBackgroundEvents() {
                     log.warn("An error occurred when processing the background event: {}", e.getMessage(), e);
             }
         }
+        kafkaAsyncConsumerMetrics.recordBackgroundEventQueueProcessingTime(time.milliseconds() - startMs);
```

Comment:
we probably don't want to record this if there were no events right?

Reply:
Good point. It'll artificially skew the averages lower.